controlplane/pkg/remote/endpoint_selector_service.go

```diff
@@ -17,31 +17,63 @@ package remote
 import (
 	"context"
 
+	"github.com/fsnotify/fsnotify"
+
 	unified "github.com/networkservicemesh/networkservicemesh/controlplane/api/connection"
 	"github.com/networkservicemesh/networkservicemesh/sdk/compat"
 
 	"github.com/pkg/errors"
 	"github.com/sirupsen/logrus"
+	"github.com/spf13/viper"
 
 	"github.com/golang/protobuf/proto"
 	"github.com/golang/protobuf/ptypes/empty"
 
 	"github.com/networkservicemesh/networkservicemesh/controlplane/api/connectioncontext"
 	unifiedconnection "github.com/networkservicemesh/networkservicemesh/controlplane/api/nsm/connection"
-	pluginapi "github.com/networkservicemesh/networkservicemesh/controlplane/api/plugins"
 	"github.com/networkservicemesh/networkservicemesh/controlplane/api/remote/connection"
 	"github.com/networkservicemesh/networkservicemesh/controlplane/api/remote/networkservice"
 	unifiednsm "github.com/networkservicemesh/networkservicemesh/controlplane/pkg/api/nsm"
 	"github.com/networkservicemesh/networkservicemesh/controlplane/pkg/common"
 	"github.com/networkservicemesh/networkservicemesh/controlplane/pkg/model"
-	"github.com/networkservicemesh/networkservicemesh/controlplane/pkg/plugins"
+	"github.com/networkservicemesh/networkservicemesh/sdk/prefix_pool"
 )
 
+type prefixHelper struct {
+	pool prefix_pool.PrefixPool
+}
+
+func NewPrefixHelper() prefixHelper {
+	ph := prefixHelper{}
+
+	empty_pool, err := prefix_pool.NewPrefixPool()
+	if err != nil {
+		logrus.Error("Failed to load excluded pool")
+	}
+	ph.pool = empty_pool
+
+	// setup watching the config file
+	v := viper.New()
+	v.SetConfigFile(prefix_pool.PrefixesFilePath)
+	v.SetDefault("excludedPrefixes", []string{})
+	v.OnConfigChange(func(e fsnotify.Event) {
+		logrus.Info("Exclude pool config file changed:", e.Name)
+		pool, err := prefix_pool.NewPrefixPool(v.GetStringSlice("excludedPrefixes")...)
```

Comment:
Is this thread safe?

Reply:
I don't see why this line should be thread safe, the assignment to ph.pool might be, though

- [ ] assign ph.pool via mutex lock